src/lib/math.ts

```
@@ -118,27 +144,20 @@ export function add(...vecs: NdVector[]): NdVector {
   return result
 }
 
-export function dot(
-  x1: NdVector,
-  x2: NdVector,
-): number | number[] | number[][] {
+export function dot(vec1: NdVector, vec2: NdVector): number | number[] | number[][] {
   // Handle the case where both inputs are scalars
-  if (typeof x1 === `number` && typeof x2 === `number`) {
-    return x1 * x2
+  if (typeof vec1 === `number` && typeof vec2 === `number`) {
```

Biome (1.9.4)

[error] Invalid `typeof` comparison value: this expression is not a string literal

not a string literal

(lint/suspicious/useValidTypeof)